Fix the EXEC_STACK in ocaml compiled binaries (#153382)

Index: ocaml-4.00.1/asmrun/sparc.S
===================================================================
--- ocaml-4.00.1.orig/asmrun/sparc.S
+++ ocaml-4.00.1/asmrun/sparc.S
@@ -358,3 +358,8 @@ caml_system__frametable:
         .type caml_raise_exception, #function
         .type caml_system__frametable, #object
 #endif
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
+
Index: ocaml-4.00.1/asmcomp/arm/emit.mlp
===================================================================
--- ocaml-4.00.1.orig/asmcomp/arm/emit.mlp
+++ ocaml-4.00.1/asmcomp/arm/emit.mlp
@@ -895,6 +895,13 @@ let data l =
   `	.data\n`;
   List.iter emit_item l
 
+(* Mark stack as non executable *)
+let nx_stack() =
+  if Config.system = "linux" then
+    `  .section .note.GNU-stack,\"\",%progbits\n`
+
+
+
 (* Beginning / end of an assembly file *)
 
 let begin_assembly() =
@@ -926,6 +933,7 @@ let begin_assembly() =
   `{emit_symbol lbl_begin}:\n`
 
 let end_assembly () =
+  nx_stack();
   let lbl_end = Compilenv.make_symbol (Some "code_end") in
   `	.text\n`;
   `	.globl	{emit_symbol lbl_end}\n`;
Index: ocaml-4.00.1/asmcomp/power/emit.mlp
===================================================================
--- ocaml-4.00.1.orig/asmcomp/power/emit.mlp
+++ ocaml-4.00.1/asmcomp/power/emit.mlp
@@ -930,6 +930,12 @@ let data l =
   emit_string data_space;
   List.iter emit_item l
 
+(* Mark stack as non executable *)
+let nx_stack() =
+  if Config.system = "elf" then
+      `  .section .note.GNU-stack,\"\",%progbits\n`
+
+
 (* Beginning / end of an assembly file *)
 
 let begin_assembly() =
@@ -946,6 +952,7 @@ let begin_assembly() =
   `{emit_symbol lbl_begin}:\n`
 
 let end_assembly() =
+  nx_stack();
   if pic_externals then
     (* Emit the pointers to external functions *)
     StringSet.iter emit_external !external_functions;
Index: ocaml-4.00.1/asmcomp/sparc/emit.mlp
===================================================================
--- ocaml-4.00.1.orig/asmcomp/sparc/emit.mlp
+++ ocaml-4.00.1/asmcomp/sparc/emit.mlp
@@ -745,6 +745,12 @@ let data l =
   `	.data\n`;
   List.iter emit_item l
 
+(* Mark stack as non executable *)
+let nx_stack() =
+  if Config.system = "linux" then
+      `  .section .note.GNU-stack,\"\",%progbits\n`
+
+
 (* Beginning / end of an assembly file *)
 
 let begin_assembly() =
@@ -758,6 +764,7 @@ let begin_assembly() =
   `{emit_symbol lbl_begin}:\n`
 
 let end_assembly() =
+  nx_stack();
   `	.text\n`;
   let lbl_end = Compilenv.make_symbol (Some "code_end") in
   `	.global	{emit_symbol lbl_end}\n`;
Index: ocaml-4.00.1/asmrun/power-elf.S
===================================================================
--- ocaml-4.00.1.orig/asmrun/power-elf.S
+++ ocaml-4.00.1/asmrun/power-elf.S
@@ -424,3 +424,8 @@ caml_system__frametable:
         .long   .L105 + 4       /* return address into callback */
         .short  -1              /* negative size count => use callback link */
         .short  0               /* no roots here */
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
+
Index: ocaml-4.00.1/asmrun/arm.S
===================================================================
--- ocaml-4.00.1.orig/asmrun/arm.S
+++ ocaml-4.00.1/asmrun/arm.S
@@ -446,3 +446,7 @@ caml_system__frametable:
         .align  2
         .type   caml_system__frametable, %object
         .size   caml_system__frametable, .-caml_system__frametable
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
