Index: ocaml-4.00.1/tools/Makefile.shared
===================================================================
--- ocaml-4.00.1.orig/tools/Makefile.shared
+++ ocaml-4.00.1/tools/Makefile.shared
@@ -268,7 +268,7 @@ beforedepend:: opnames.ml
 # Display info on compiled files
 
 objinfo_helper$(EXE): objinfo_helper.c ../config/s.h
-	$(BYTECC) -o objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
+	$(MKEXE) -o objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
           objinfo_helper.c $(LIBBFD_LINK)
 
 OBJINFO=../utils/misc.cmo ../utils/config.cmo \
