Index: ocaml-4.05.0-beta2/tools/Makefile
===================================================================
--- ocaml-4.05.0-beta2.orig/tools/Makefile
+++ ocaml-4.05.0-beta2/tools/Makefile
@@ -320,7 +320,7 @@ CCOUT = -o $(EMPTY)
 endif
 
 objinfo_helper$(EXE): objinfo_helper.c ../config/s.h
-	$(BYTECC) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
+	$(MKEXE) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
           $(DEF_SYMBOL_PREFIX) $(LIBBFD_INCLUDE) objinfo_helper.c $(LIBBFD_LINK)
 
 OBJINFO=../compilerlibs/ocamlcommon.cma \
